alchemy/src/neon/endpoint.ts

```diff
@@ -0,0 +1,626 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { handleApiError } from "./api-error.ts";
+import { createNeonApi, type NeonApiOptions, type NeonApi } from "./api.ts";
+import type { NeonProject } from "./project.ts";
+import type { NeonBranch } from "./branch.ts";
+
+/**
+ * Properties for creating or updating a Neon endpoint
+ */
+export interface NeonEndpointProps extends NeonApiOptions {
+  /**
+   * The project containing the branch
+   */
+  project: string | NeonProject;
+
+  /**
+   * The branch where the endpoint will be created
+   */
+  branch: string | NeonBranch;
+
+  /**
+   * Type of endpoint (read-write or read-only)
+   */
+  type: "read_write" | "read_only";
+
+  /**
+   * Compute provisioner to use
+   */
+  computeProvisioner?: "k8s-pod" | "k8s-neonvm";
+
+  /**
+   * PostgreSQL settings for the endpoint
+   */
+  settings?: {
+    pgSettings?: Record<string, string>;
+  };
+
+  /**
+   * Whether connection pooling is enabled
+   */
+  poolerEnabled?: boolean;
+
+  /**
+   * Connection pooling mode
+   */
+  poolerMode?: "session" | "transaction";
+
+  /**
+   * Whether the endpoint is disabled
+   */
+  disabled?: boolean;
+
+  /**
+   * Whether passwordless access is enabled
+   */
+  passwordlessAccess?: boolean;
+
+  /**
+   * Timeout in seconds before the endpoint suspends
+   */
+  suspendTimeoutSeconds?: number;
+
+  /**
+   * Provisioner type
+   */
+  provisioner?: "k8s-pod" | "k8s-neonvm";
+
+  /**
+   * Region where the endpoint should be created
+   */
+  regionId?: string;
```

Comment:
Should this be a string union type src/neon/region.ts and shared across all resources?